### PR TITLE
F/syslog parse date speed up

### PR DESCRIPTION
**Pull request by bkil-syslogng**
Resolve some of the bottlenecks when processing syslog source, 2-3x speedup.

Some more bottlenecks remain to be investigated, but the implemented parts (existing commits) can already be reviewed.

### CODE REVIEW

**Comment by bazsi** [comment]
Nice refactor. 

FYI: The LP_XXX flags are an ugly layering violation, at one point we should probably fix that part up, but this patch is clearly a step forward. so nice one, :+1:

File: .travis.yml [hunk 1]
```diff
@@ -94,7 +95,7 @@ matrix:
           astyle
       before_script:
       script:
-      - scripts/style-checker.sh check
+      - scripts/style-checker.sh diff
```
Comment (lbudai) [inline .travis.yml]: Do we really want to put the diff in the travis log?

Comment (bkil-syslogng) [inline .travis.yml]: Yes, that was what we have agreed upon some weeks ago, wasn't it? We could
do fine tuning of diff options, though.

It helped me fix the issues detected in this case for example.

**Comment by juhaszviktor** [comment]
The travis/stylechecker patch is not in this context

File: scripts/style-checker.sh [hunk 2]
```diff
@@ -52,12 +52,35 @@ function astyle_c_format
     exit 0
 }
 
+function astyle_c_diff
+{
+    astyle --options="$root_dir/.astylerc" "$root_dir/*.c" |
+    grep "Formatted" |
```
Comment (kvch) [inline scripts/style-checker.sh]: Why are these lines needed? Why is simply calling `git diff` not enough?

File: scripts/style-checker.sh [hunk 3]
```diff
@@ -52,12 +52,35 @@ function astyle_c_format
     exit 0
 }
 
+function astyle_c_diff
```
Comment (kvch) [inline scripts/style-checker.sh]: AFAIK we decided that diff view is unecessary.

File: .travis.yml (hunk 1 above)
Comment (kvch) [inline .travis.yml]: First we thought it was a good idea. But than, it was discarded.

File: modules/syslogformat/syslog-format.c [hunk 4]
```diff
@@ -437,28 +435,29 @@ log_msg_parse_date(LogMessage *self, const guchar **data, gint *length, guint pa
...
           src++;
           left--;
         }
       else if (G_UNLIKELY(src[0] == '.'))
         {
-          log_msg_set_value(self, is_synced, "1", 1);
+          if (!(parse_flags & LP_NO_PARSE_DATE))
+            log_msg_set_value(self, is_synced, "1", 1);
           src++;
           left--;
```
Comment (juhaszviktor) [inline modules/syslogformat/syslog-format.c]: We should factor out the setting of the is_synced, shouldn't we?

File: scripts/style-checker.sh (hunk 2 above)
Comment (bkil-syslogng) [inline scripts/style-checker.sh]: I don't understand the question. Which lines? Enough for what? Calling it when after what?

File: .travis.yml (hunk 1 above)
Comment (bkil-syslogng) [inline .travis.yml]: Not sure when it was discarded, it was accepted on the meetings that I have
attended.

I still think it is a very useful feature. Giving me a red exclamation mark
without any detailed reason can put me off. Copyright checker also gives
hints about the fixes. gcc gives a hint about what an error means, it
doesn't just say that one of the specified files is in error.

**Comment by bkil-syslogng** [comment]
You mean we should split off the style checker patch from this PR?

File: scripts/style-checker.sh (hunk 2 above)
Comment (kvch) [inline scripts/style-checker.sh]: I mean lines 58-65. AFAIK the sources are formatted, so those files are changed. This, it is enough to simply call `git diff` after formatting files using `astyle`.

**Comment by bkil-syslogng** [comment]
I've removed the [style checker patch](https://github.com/bkil-syslogng/syslog-ng/commit/9dad1cb2d4e02d302fbc9a7bffc772606179235d) from this PR, but it should be merged as well soon as per above.
